lib/engine/game/base.rb

```diff
@@ -1051,31 +1051,41 @@ def place_home_token(corporation)
         city.place_token(corporation, token)
       end
 
-      def tile_cost(tile, hex, entity)
+      def upgrade_cost(tile, hex, entity)
```

Comment:
why is this renamed?

Reply:
I found the function name misleading. The actual tile cost in the step is a much longer calculation involving tile lay abilities, borders, extra costs, etc. All this function does is calculate and log the discounts for _upgrades_ for the newly placed tile, hence the name change.